src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice.cs

```diff
@@ -0,0 +1,176 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Jint.Native;
+using Nethermind.Blockchain.Find;
+using Nethermind.Core;
+using Nethermind.Int256;
+using Nethermind.TxPool;
+
+namespace Nethermind.JsonRpc.Modules.Eth
+{
+    public class GasPrice
+    {
+        private static UInt256? _lastPrice = null; //will this be okay if it is static?
+        private static Block? _lastHeadBlock = null;
+        private readonly IBlockFinder _blockFinder;
+        public GasPrice(IBlockFinder blockFinder)
+        {
+            _blockFinder = blockFinder;
+        }
+        private bool eth_addTxFromBlockToSet(Block block, ref SortedSet<UInt256> sortedSet, UInt256 finalPrice,
+            UInt256? ignoreUnder = null, long? maxCount = null)
+        {
+            ignoreUnder ??= UInt256.Zero;
+            int length = block.Transactions.Length;
+            int added = 0;
+            if (length > 0)
+            {
+                for (int index = 0; index < length && (maxCount == null || sortedSet.Count < maxCount); index++)
+                {
+                    Transaction transaction = block.Transactions[index];
+                    if (!transaction.IsEip1559 && transaction.GasPrice >= ignoreUnder)
+                    {
+                        sortedSet.Add(transaction.GasPrice);
+                        added++;
+                    }
+                }
+
+                switch (added)
+                {
+                    case 0:
+                        sortedSet.Add(finalPrice);
+                        return (false);
+                    case 1:
+                        return (false);
+                    default:
+                        return true;
+                }
+            }
+            else
+            {
+                sortedSet.Add(finalPrice);
+                return false;
+            }
+        }
+
+        private UInt256? eth_latestGasPrice(long headBlockNumber, long genesisBlockNumber)
+        {
+            while (headBlockNumber >= genesisBlockNumber) //should this be latestBlock or headBlock?
+            {
+                Transaction[] transactions = _blockFinder.FindBlock(headBlockNumber)!.Transactions
+                    .Where(t => !t.IsEip1559).ToArray();
+                if (transactions.Length > 0)
+                {
+                    return transactions[^1].GasPrice;
```

Comment:
this is minimum gas price from previous non-empty block, did not you want to go slightly above this?

Reply:
From what I understood, I just saw Transactions was a regular array of transactions, so I thought the last element from the last non-empty block would be the latest added transaction and would have the "latestPrice" (which is what Geth uses). 

Of course, this is not set in stone - would you like me to try to get the 3rd or the 4th latest transaction instead?